client/src/components/assessments/AssessmentResultsTable.js

```diff
@@ -0,0 +1,325 @@
+import AggregationWidget from "components/AggregationWidgets"
+import AddAssessmentModal from "components/assessments/AddAssessmentModal"
+import { ReadonlyCustomFields } from "components/CustomFields"
+import Fieldset from "components/Fieldset"
+import { Formik } from "formik"
+import LinkTo from "components/LinkTo"
+import _isEmpty from "lodash/isEmpty"
+import PropTypes from "prop-types"
+import React, { useState } from "react"
+import { Button, Table } from "react-bootstrap"
+import "components/assessments/AssessmentResultsTable.css"
+
+/* The AssessmentResultsTable component displays the results of two types of
+ * assessments made on a given entity and subentities:
+ * - aggregation of the measurements made on the entity/subentities when
+ *   working on them in relation to another type of entity (example:
+ *   assessments made on tasks, while filling  report related to the tasks);
+ *   the definition of these assessments is to be found in
+ *   entity.customFields.assessmentDefinition
+ * - display of the last monthly assessment made on the entity/subentities
+ *   as a conclusion about the given period of time;
+ *   the definition of these assessments is to be found in
+ *   entity.periodAssessmentConfig()
+ */
+
+const AssessmentsTableHeader = ({ periods }) => (
+  <thead>
+    <tr key="periods">
+      <>
+        {periods.map(period => (
+          <th key={period.start}>{period.start.format("MMM-YYYY")}</th>
+        ))}
+      </>
+    </tr>
+  </thead>
+)
+AssessmentsTableHeader.propTypes = {
+  periods: PropTypes.array
+}
+
+const MeasurementRow = ({
+  measurementKey,
+  measurementDef,
+  entity,
+  assessmentPeriods
+}) => {
+  const aggWidgetProps = {
+    widget: measurementDef.aggregation?.widget || measurementDef.widget,
+    aggregationType: measurementDef.aggregation?.aggregationType,
+    vertical: true
+  }
+  const widgetLayoutConfig = Object.without(
+    measurementDef,
+    "aggregation",
+    "type",
+    "typeError",
+    "placeholder",
+    "helpText",
+    "validations",
+    "visibleWhen",
+    "objectFields"
+  )
+  return (
+    <tr>
+      {assessmentPeriods.map((assessmentPeriod, index) => (
+        <td key={index}>
```

Comment:
This appears to have a variable number of `td`s; are you sure each row (here and below) has the same number of `td`s?

Reply:
They are actually all looping through the assessmentPeriods (directly or indirectly), so yes, they should have the same amount of td's.

Reply:
OK; it wasn't immediately obvious to me that `assessmentPeriods` and `periodsLastAssessment` always have the same number of elements.